src/HealthChecks.SendGrid/SendGridHealthCheck.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace HealthChecks.SendGrid
+{
+    public class SendGridHealthCheck : IHealthCheck
+    {
+        private const string MailAddressName = "Health Check User";
+        private const string MailAddress = "healthcheck@example.com";
+        private const string Subject = "Checking health is Fun";
+        private readonly string _apiKey;
+        private readonly Func<HttpClient> _httpClientFactory;
+
+        public SendGridHealthCheck(string apiKey, Func<HttpClient> httpClientFactory)
+        {
+            _apiKey = apiKey ?? throw new ArgumentNullException(nameof(apiKey));
+            _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var httpClient = _httpClientFactory();
+                var client = new SendGridClient(httpClient, _apiKey);
+                var from = new EmailAddress(MailAddress, MailAddressName);
+                var to = new EmailAddress(MailAddress, MailAddressName);
+                var msg = MailHelper.CreateSingleEmail(from, to, Subject, Subject, null);
+                msg.SetSandBoxMode(true);
+                var response = await client.SendEmailAsync(msg, cancellationToken);
+
+                HealthCheckResult result;
+                var isHealthy = response.StatusCode == HttpStatusCode.OK;
```

Comment:
is more natural, if ( response.IsSuccessStatusCode ) instead this comparasion

Reply:
Can you reduce if/else

```csharp
if (!response.IsSuccessStatusCode)
{
return new HealthCheckResult( ..... )
}
return HealthCheckResult.Healty()
```

Reply:
Unfortunately this can not be done, because `await client.SendEmailAsync(msg, cancellationToken)` returns the type `Task<Response>` and `Response` is a SendGrip C# [specific type](https://github.com/sendgrid/sendgrid-csharp/blob/94e94f36570c62153f4bb43eab8c27ac90fef094/src/SendGrid/Response.cs).

Also the [SendGrid documentation](https://sendgrid.com/docs/for-developers/sending-email/sandbox-mode/#valid-request-body) explicitly states that a `200` is returned. 
Whereas `IsSuccessStatusCode` is 

```CSharp
        public bool IsSuccessStatusCode
        {
            get { return ((int)_statusCode >= 200) && ((int)_statusCode <= 299); }
        }
```
 Reference [dotnet/runtime](https://github.com/dotnet/runtime/blob/4b03513e20e0caf9ee34817eb74356903f1bb33e/src/libraries/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs#L143-L146
)

Reply:
IsSuccessStatusCode include 200 and all 200-299 are Success response, I think is more natural!